content/frameworks/react.textile

```diff
@@ -0,0 +1,177 @@
+---
+title: React
+meta_description: "The React submodule enables you to use React Hooks to connect to Ably."
+language:
+  - react
+---
+
+Leverage the power of Ably in your React applications using idiomatic, easy-to-use React Hooks. This package allows you to:
+
+* Interact with Ably channels using a react hook.
+* Send messages via Ably using the channel instances the hooks provide.
+* Get notifications of user presence on channels.
+* Send presence updates.
+
+
+Available hooks:
+
+- "useChannel":#useChannel := The @useChannel@ hook subscribes to a channel and receives messages from it.
+- "usePresence":#usePresence := The @usePresence@ hook subscribes to presence events on a channel, notifying when users join or leave the channel.
+- "useConnectionStateListener":#useConnectionStateListener := The @useConnectionStateListener@ hook attaches a listener to be notified of connection state changes in the Ably client.
+- "useChannelStateListener":#useChannelStateListener := The @useChannelStateListener@ hook attaches a listener to be notified of channel state changes.
+- "useAbly":#useAbly := The @useAbly@ hook grants access to the Ably client instance provided by the AblyProvider context.
+
+These hooks manage the lifecycle of the Ably SDK instances for you, ensuring that you subscribe and unsubscribe to channels and events when your React components re-render.
+
+h2(#installation). Installation
+
+To install the Ably React Hooks, use the following command:
+
+```[sh]
+npm install --save ably
+```
+
+**Note:** You will need React 16.8 or above to use these hooks.
+
+h2(#set-up-ably). Set up Ably
+
+To utilize these hooks, you'll need an Ably API key. If you haven't signed up yet, sign up now for a free Ably account. Once registered:
+
+* Log into your app dashboard.
+* Under "Your apps", click on "Manage app" for any app you wish to use for this tutorial, or create a new one with the "Create New App" button.
+* Navigate to the "API Keys" tab.
+* Copy the secret "API Key" value from your Root key. This will be used later when building your app.
+
+**Note:** It's recommended to use Token Authentication. This requires server-side code, which is beyond the scope of this documentation. In the examples provided, we use an API key directly in the markup. This approach is for local development only and should not be used in production. Ensure you don't commit this to your repositories.
+
+h2(#usage). Usage
+
+h3(#ably-provider). Setting up the Ably Provider
+
+To connect your app to Ably, use the @AblyProvider@ component. This component should be placed high in your component tree, wrapping every component that needs to access Ably.
+
+```[react]
+import * as Ably from 'ably';
+import { AblyProvider, useChannel, usePresence } from 'ably/react';
+
+const client = {
+  key: "your-ably-api-key",
+  clientId: "your-ably-client-id",
+}
+
+root.render(
+  <AblyProvider client={client}>
```

Comment:
This example is slightly wrong - you can't pass a `ClientOptions` in as the `client` prop, the prop (and the variable) should be called `options` instead. I also think the primary way the provider should be used is by passing in a client instance so I think that should be the first example and then `options` should be below it as an alternative.

Reply:
Done (hopefully correctly) ✔️